### PR TITLE
[IMP] Fleet: driver end date activity

### PR DESCRIPTION
**Pull request by kp-merucore**
PURPOSE

Currently, when changing the driver of a vehicle, we don't set the end date
of the last driver and we can't automatically set it because this end
date can be another date that the start of the new driver.

SPECIFICATIONS

When we change the driver of a vehicle, add an activity to specify
the end date of the last driver.

LINKS

PR https://github.com/odoo/odoo/pull/57901

Task-2328981

### CODE REVIEW

File: addons/fleet/models/fleet_vehicle.py [hunk 1]
```diff
@@ -218,6 +218,11 @@ def write(self, vals):
         if 'driver_id' in vals and vals['driver_id']:
             driver_id = vals['driver_id']
             self.filtered(lambda v: v.driver_id.id != driver_id).create_driver_history(driver_id)
+            note = _('Specify the End date of %s') % self.driver_id.name
+            self.activity_schedule(
+                'mail.mail_activity_data_todo',
+                user_id=self.manager_id.id or self.env.user.id,
```
Comment (jbm-odoo) [inline addons/fleet/models/fleet_vehicle.py:224]: self can be multi, you can put it in a for loop (write a driver on multi record is not common, but you must manage this use case)

File: addons/fleet/models/fleet_vehicle.py [hunk 2]
```diff
@@ -218,6 +218,11 @@ def write(self, vals):
...
+                note=note)
```
Comment (jbm-odoo) [inline addons/fleet/models/fleet_vehicle.py:225]: Don't need to create this activity if they are no old driver

**Review by jbm-odoo — COMMENTED** [review]
As it isn't a bug fix, you can target master instead of 13.0

**Comment by kat-odoo** [comment]
Target Master

**Pull request closed without merging**